ebill_paynet/views/ebill_payment_contract.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record model="ir.ui.view" id="ebill_payment_contract_form_view">
+        <field name="name">ebill.payment.contract.form (in ebill_panyet)</field>
+        <field name="model">ebill.payment.contract</field>
+        <field
+            name="inherit_id"
+            ref="base_ebill_payment_contract.ebill_payment_contract_form_view"
+        />
+        <field name="arch" type="xml">
+            <field name="partner_id" position="after">
+                <field
+                    name="paynet_account_number"
+                    attrs="{'invisible': [('is_paynet_contract', '=', False)]}"
+                />
+                <field
+                    name="paynet_service_id"
+                    attrs="{'invisible': [('is_paynet_contract', '=', False)]}"
+                />
+                <field name="is_paynet_contract" invisible="1" />
+            </field>
```

Comment:
I would put this in a separate group as `base_ebill_payment_contract` module could be used outside of Paynet context I suppose, so based on my comment on https://github.com/OCA/edi/pull/207#discussion_r475585357:
```suggestion
        <group name="base" position="after">
            <group name="paynet" position="after">
                <field
                    name="paynet_account_number"
                    attrs="{'invisible': [('is_paynet_contract', '=', False)]}"
                />
                <field
                    name="paynet_service_id"
                    attrs="{'invisible': [('is_paynet_contract', '=', False)]}"
                />
                <field name="is_paynet_contract" invisible="1" />
            </group>
        </group>
```

Reply:
Much better done